alpharaw/match/psm_match.py

```diff
@@ -73,17 +73,41 @@ def __init__(
         self.use_ppm = use_ppm
         self.tolerance = tol_value
 
-    def _preprocess_psms(self, psm_df):
-        pass
+    def get_fragment_mz_df(self) -> pd.DataFrame:
+        """
+        Call :func:`alphabase.peptide.fragment.create_fragment_mz_dataframe`
+        for :attr:`PepSpecMatch.psm_df` and :attr:`PepSpecMatch.charged_frag_types`.
 
-    def get_fragment_mz_df(self):
+
+        Returns
+        -------
+        DataFrame
+            _description_
+        """
         return create_fragment_mz_dataframe(
             self.psm_df,
             self.charged_frag_types,
             dtype=PEAK_MZ_DTYPE,
         )
 
-    def _add_missing_columns_to_psm_df(self, psm_df: pd.DataFrame, raw_data=None):
+    def _add_missing_columns_to_psm_df(
+        self, psm_df: pd.DataFrame, raw_data: MSData_Base = None
+    ):
+        """
+        Add missing "rt", "nce", "rt_norm", ("mobility") columns to `psm_df` if missing.
+
+        Parameters
+        ----------
+        psm_df : pd.DataFrame
+            psm dataframe to be processed.
+        raw_data : MSData_Base, optional
+            The `MSData_Base`. If None, `self.raw_data`. by default None.
+
+        Returns
+        -------
+        DataFrame
+            psm_df inplace.
```

Comment:
as far as I can see, there's no in-place mod of psm_df here?

Reply:
We add columns into psm_df, in `_add_columns_if_missing` function